### PR TITLE
Moves almost all wrench checks in attackby to wrench_act()

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Does a pass over all the instances that I could find of things like `if(istype(thing, /obj/item/wrench))` and converted them to `wrench_act()` calls. I'd ideally like to do this with other different checks, if possible, but this is a definite one thing at a time kind of deal.
I left one or two in place that are used for construction, mostly because I didn't want to break up the steps toooo much.
This also converts all of the `is[tool]()` procs to check tool behavior. 

## Why It's Good For The Game
If we can eventually move out to tool act calls, then we can be a *lot* more creative with how we handle tools, particularly by letting tools be defined by `tool_behaviour` instead of just being a subtype of the base tool. I'm coming for you, power drill.

## Testing
It compiles, I need to verify that everything works as it did before.
WIP

## Changelog

NPFC

### CODE REVIEW

File: code/game/machinery/pipe/pipe_construction.dm [hunk 1]
```diff
@@ -537,18 +537,16 @@
...
-
+/obj/item/pipe_meter/wrench_act(mob/living/user, obj/item/I)
 	if(!locate(/obj/machinery/atmospherics/pipe, loc))
-		to_chat(user, "<span class='warning'>You need to fasten it to a pipe</span>")
+		to_chat(user, "<span class='warning'>You need to fasten it to a pipe.</span>")
 		return TRUE
 
 	new /obj/machinery/atmospherics/meter(loc)
-	playsound(loc, W.usesound, 50, 1)
+	W.play_tool_sound(src)
```
Comment (DGamerL) [inline code/game/machinery/pipe/pipe_construction.dm:546]:
```suggestion
	I.play_tool_sound(src)
```

Comment (lewcc) [reply code/game/machinery/pipe/pipe_construction.dm]: I swear I tested this, but in my defense, I was on an airplane

File: code/game/objects/structures/engicart.dm [hunk 2]
```diff
@@ -82,24 +82,36 @@
...
+		user.visible_message(
+			"[user] loosens [src]'s casters.",
+			"<span class='notice'> You have loosened [src]'s casters.</span>",
+			"You hear ratchet."
+		)
+		anchored = FALSE
+
+	return TRUE
+
+
```
Comment (DGamerL) [inline code/game/objects/structures/engicart.dm:114]:
```suggestion
```

File: code/game/objects/structures/kitchen_spike.dm [hunk 3]
```diff
@@ -10,17 +10,19 @@
 	anchored = TRUE
 	max_integrity = 200
 
+/obj/structure/kitchenspike_frame/wrench_act(mob/living/user, obj/item/I)
+	if(!I.tool_use_check(user, 0))
+		return
```
Comment (DGamerL) [inline code/game/objects/structures/kitchen_spike.dm:15]:
```suggestion
		return FALSE
```

File: code/game/turfs/simulated/floor/plating.dm [hunk 4]
```diff
@@ -206,17 +206,16 @@
 	acidpwr = min(acidpwr, 50) //we reduce the power so reinf floor never get melted.
 	. = ..()
 
-/turf/simulated/floor/engine/attackby(obj/item/C as obj, mob/user as mob, params)
-	if(!C || !user)
+/turf/simulated/floor/engine/wrench_act(mob/living/user, obj/item/wrench/W)
+	if(!istype(W) || !user)
 		return
```
Comment (DGamerL) [inline code/game/turfs/simulated/floor/plating.dm:211]: Do we need this `istype()` here?

File: code/modules/atmospherics/machinery/atmospherics.dm [hunk 5]
```diff
@@ -187,63 +190,71 @@ Pipelines + Other Objects -> Pipe network
...
-					unsafe_pressure_release(user,internal_pressure)
-			deconstruct(TRUE)
-	else
-		if(T.transparent_floor)
-			to_chat(user, "<span class='danger'>You can't interact with something that's under the floor!</span>")
-			return TRUE
-		return ..()
+				unsafe_pressure_release(user,internal_pressure)
+		deconstruct(TRUE)
+
```
Comment (DGamerL) [inline code/modules/atmospherics/machinery/atmospherics.dm:249]:
```suggestion
```

File: code/modules/mob/living/silicon/ai/ai_mob.dm [hunk 6]
```diff
@@ -1249,27 +1249,22 @@ GLOBAL_LIST_INIT(ai_verbs_default, list(
...
 			return
-	else
-		return ..()
+		user.visible_message("<span class='notice'>[user] finishes unfastening [src]!</span>")
+		anchored = FALSE
+		return
+	user.visible_message("<span class='notice'>[user] starts to bolt [src] to the plating...</span>")
+	if(!I.use_tool(src, user, 4 SECONDS, 0, 50))
+		user.visible_message("<span class='notice'>[user] decides not to bolt [src].</span>")
+		return
```
Comment (DGamerL) [inline code/modules/mob/living/silicon/ai/ai_mob.dm:1265]:
```suggestion
		return FALSE
```

File: code/modules/power/engines/singularity/collector.dm [hunk 7]
```diff
@@ -47,6 +47,24 @@
 		else
 			to_chat(user, "<span class='warning'>The controls are locked!</span>")
 
+/obj/machinery/power/rad_collector/wrench_act(mob/living/user, obj/item/I)
+	. = ..()
```
Comment (DGamerL) [inline code/modules/power/engines/singularity/collector.dm:51]: Does this need to call parent? Pretty much no other `wrench_act` does that

File: code/game/turfs/simulated/floor/plating.dm (hunk 4 above)
Comment (lewcc) [reply code/game/turfs/simulated/floor/plating.dm:211]: Yeah, probably not; good catch!

File: code/game/objects/structures/fluff.dm [hunk 8]
```diff
@@ -10,17 +10,18 @@
...
-			new /obj/item/stack/sheet/metal(drop_location())
-			qdel(src)
-		return
-	return ..()
+/obj/structure/fluff/wrench_act(mob/living/user, obj/item/I)
+	if(!deconstructible)
+		return FALSE
+
+	user.visible_message("<span class='notice'>[user] starts disassembling [src]...</span>", "<span class='notice'>You start disassembling [src]...</span>")
+	playsound(loc, I.usesound, 50, 1)
```
Comment (Burzah) [inline code/game/objects/structures/fluff.dm:18]:
```suggestion
	playsound(loc, I.usesound, 50, TRUE)
```

File: code/game/objects/structures/fluff.dm [hunk 9]
```diff
@@ -10,17 +10,18 @@
...
+	if(I.use_tool(src, user, 5 SECONDS, 0, 50))
+		user.visible_message("<span class='notice'>[user] disassembles [src]!</span>", "<span class='notice'>You break down [src] into scrap metal.</span>")
+		playsound(user, 'sound/items/deconstruct.ogg', 50, 1)
```
Comment (Burzah) [inline code/game/objects/structures/fluff.dm:21]:
```suggestion
		playsound(user, 'sound/items/deconstruct.ogg', 50, TRUE)
```

File: code/modules/recycling/disposal-construction.dm [hunk 10]
```diff
@@ -137,10 +137,41 @@
...
+			density = TRUE
+		to_chat(user, "You detach the [nicetype] from the underfloor.")
+	else
+		anchored = TRUE
+		if(ispipe)
+			level = 1 // We don't want disposal bins to disappear under the floors
+			density = FALSE
+		else
+			density = TRUE // We don't want disposal bins or outlets to go density 0
+		to_chat(user, "You attach the [nicetype] to the underfloor.")
```
Comment (Burzah) [inline code/modules/recycling/disposal-construction.dm:158]: Should these two be using some type of span class? Similar tool to chat messages are using `notice` as the default.

**Comment by SteelSlayer** [comment]
This went on TM for one round. We found that this PR makes un-wrenching pipes impossible which is pretty inconvenient for engineering. Pulling it until that's fixed.

File: code/modules/recycling/disposal-construction.dm (hunk 10 above)
Comment (lewcc) [reply code/modules/recycling/disposal-construction.dm:158]: They should, disposals code is just so dusty and old

File: code/game/machinery/mass_driver.dm [hunk 11]
```diff
@@ -142,7 +142,7 @@
 			return FALSE
 
 		if(4) // Grille in place
-			if(W.tool_behaviour == TOOL_CROWBAR)
+			if(iscrowbar(W))
 				to_chat(user, "You begin to pry off the grille from \the [src]...")
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:146]:
```suggestion
				to_chat(user, "<span class='notice'>You begin to pry off the grille from \the [src]...</span>")
```

File: code/game/machinery/mass_driver.dm [hunk 12]
```diff
@@ -120,7 +120,7 @@
 			return FALSE
 
 		if(3) // Wired
-			if(istype(W, /obj/item/wirecutters))
+			if(iswirecutter(W))
 				to_chat(user, "You begin to remove the wiring from \the [src].")
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:124]:
```suggestion
				to_chat(user, "<span class='notice'>You begin to remove the wiring from \the [src].</span>")
```

File: code/game/machinery/mass_driver.dm [hunk 13]
```diff
@@ -97,7 +97,7 @@
 			return FALSE
 
 		if(1) // Fixed to the floor
-			if(istype(W, /obj/item/wrench))
+			if(iswrench(W))
 				to_chat(user, "You begin to de-anchor \the [src] from the floor.")
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:101]:
```suggestion
				to_chat(user, "<span class='notice'>You begin to de-anchor \the [src] from the floor.</span>")
```

File: code/game/machinery/mass_driver.dm [hunk 14]
```diff
@@ -86,7 +86,7 @@
 /obj/machinery/mass_driver_frame/attackby(obj/item/W as obj, mob/user as mob)
 	switch(build)
 		if(0) // Loose frame
-			if(istype(W, /obj/item/wrench))
+			if(iswrench(W))
 				to_chat(user, "You begin to anchor \the [src] on the floor.")
 				playsound(get_turf(src), W.usesound, 50, 1)
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:91]:
```suggestion
				playsound(get_turf(src), W.usesound, 50, TRUE)
```

File: code/game/machinery/mass_driver.dm [hunk 15]
```diff
@@ -86,7 +86,7 @@
 /obj/machinery/mass_driver_frame/attackby(obj/item/W as obj, mob/user as mob)
 	switch(build)
 		if(0) // Loose frame
-			if(istype(W, /obj/item/wrench))
+			if(iswrench(W))
 				to_chat(user, "You begin to anchor \the [src] on the floor.")
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:90]:
```suggestion
				to_chat(user, "<span class='notice'>You begin to anchor \the [src] on the floor.</span>")
```

File: code/game/machinery/mass_driver.dm [hunk 16]
```diff
@@ -108,7 +108,7 @@
 			return FALSE
 
 		if(2) // Welded to the floor
-			if(istype(W, /obj/item/stack/cable_coil))
+			if(iscoil(W))
 				var/obj/item/stack/cable_coil/C = W
 				to_chat(user, "You start adding cables to \the [src]...")
 				playsound(get_turf(src), C.usesound, 50, 1)
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:114]:
```suggestion
				to_chat(user, "<span class='notice'>You start adding cables to \the [src]...</span>")
				playsound(get_turf(src), C.usesound, 50, TRUE)
```

File: code/game/machinery/mass_driver.dm [hunk 17]
```diff
@@ -142,7 +142,7 @@
 			return FALSE
 
 		if(4) // Grille in place
-			if(W.tool_behaviour == TOOL_CROWBAR)
+			if(iscrowbar(W))
 				to_chat(user, "You begin to pry off the grille from \the [src]...")
 				playsound(get_turf(src), W.usesound, 50, 1)
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:147]:
```suggestion
				playsound(get_turf(src), W.usesound, 50, TRUE)
```

File: code/game/machinery/mass_driver.dm [hunk 18]
```diff
@@ -120,7 +120,7 @@
 			return FALSE
 
 		if(3) // Wired
-			if(istype(W, /obj/item/wirecutters))
+			if(iswirecutter(W))
 				to_chat(user, "You begin to remove the wiring from \the [src].")
 				if(do_after(user, 10 * W.toolspeed, target = src) && (build == 3))
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:125]:
```suggestion
				if(do_after(user, 1 SECONDS * W.toolspeed, target = src) && (build == 3))
```

File: code/game/machinery/mass_driver.dm [hunk 19]
```diff
@@ -97,7 +97,7 @@
 			return FALSE
 
 		if(1) // Fixed to the floor
-			if(istype(W, /obj/item/wrench))
+			if(iswrench(W))
 				to_chat(user, "You begin to de-anchor \the [src] from the floor.")
 				playsound(get_turf(src), W.usesound, 50, 1)
 				if(do_after(user, 10 * W.toolspeed, target = src) && (build == 1))
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:103]:
```suggestion
				playsound(get_turf(src), W.usesound, 50, TRUE)
				if(do_after(user, 1 SECONDS * W.toolspeed, target = src) && (build == 1))
```

File: code/game/machinery/mass_driver.dm [hunk 20]
```diff
@@ -86,7 +86,7 @@
 /obj/machinery/mass_driver_frame/attackby(obj/item/W as obj, mob/user as mob)
 	switch(build)
 		if(0) // Loose frame
-			if(istype(W, /obj/item/wrench))
+			if(iswrench(W))
 				to_chat(user, "You begin to anchor \the [src] on the floor.")
 				playsound(get_turf(src), W.usesound, 50, 1)
 				if(do_after(user, 10 * W.toolspeed, target = src) && (build == 0))
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:92]:
```suggestion
				if(do_after(user, 1 SECONDS * W.toolspeed, target = src) && (build == 0))
```

File: code/game/machinery/mass_driver.dm [hunk 21]
```diff
@@ -120,7 +120,7 @@
 			return FALSE
 
 		if(3) // Wired
-			if(istype(W, /obj/item/wirecutters))
+			if(iswirecutter(W))
 				to_chat(user, "You begin to remove the wiring from \the [src].")
 				if(do_after(user, 10 * W.toolspeed, target = src) && (build == 3))
 					new /obj/item/stack/cable_coil(loc,2)
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:126]:
```suggestion
					new /obj/item/stack/cable_coil(loc, 2)
```

File: code/game/machinery/mass_driver.dm [hunk 22]
```diff
@@ -142,7 +142,7 @@
 			return FALSE
 
 		if(4) // Grille in place
-			if(W.tool_behaviour == TOOL_CROWBAR)
+			if(iscrowbar(W))
 				to_chat(user, "You begin to pry off the grille from \the [src]...")
 				playsound(get_turf(src), W.usesound, 50, 1)
 				if(do_after(user, 30 * W.toolspeed, target = src) && (build == 4))
```
Comment (Burzah) [inline code/game/machinery/mass_driver.dm:148]:
```suggestion
				if(do_after(user, 3 SECONDS * W.toolspeed, target = src) && (build == 4))
```

**Review by Burzah — COMMENTED** [review]
Review bomb. Might as well fix these while we are at it.

File: code/modules/recycling/disposal-construction.dm (hunk 10 above)
Comment (Burzah) [reply code/modules/recycling/disposal-construction.dm:158]: Well... I left you some reviews. 😄

**Review by Sirryan2002 — APPROVED** [review]

**Review by Burzah — COMMENTED** [review]
Since this has been on TM for a bit, how do you feel it has performed? If everything is good, can you update the testing section, please?

File: code/__HELPERS/tool_helpers.dm [hunk 23]
```diff
@@ -1,35 +1,27 @@
 //Quick type checks for some tools
 
 // Why are these not defines?
```
Comment (DGamerL) [inline code/__HELPERS/tool_helpers.dm:3]: Maybe sometime in the future 😔

File: code/modules/recycling/disposal-construction.dm [hunk 24]
```diff
@@ -137,10 +137,41 @@
...
+	var/ispipe = is_pipe()
+	var/nicetype = get_nice_name()
+	if(anchored)
+		anchored = FALSE
+		if(ispipe)
+			level = 2
+			density = FALSE
+		else
+			density = TRUE
+		to_chat(user, "You detach the [nicetype] from the underfloor.")
```
Comment (DGamerL) [inline code/modules/recycling/disposal-construction.dm:150]:
```suggestion
		to_chat(user, "<span class='notice'>You detach the [nicetype] from the underfloor.</span>")
```

File: code/modules/recycling/disposal-construction.dm (hunk 10 above)
Comment (DGamerL) [inline code/modules/recycling/disposal-construction.dm:158]:
```suggestion
		to_chat(<span class='notice'>user, "You attach the [nicetype] to the underfloor.</span>")
```

File: code/modules/recycling/disposal-construction.dm [hunk 25]
```diff
@@ -150,38 +181,19 @@
...
-			ispipe = 1
-		else
-			nicetype = "pipe"
-			ispipe = 1
+	return nicetype
+
+/obj/structure/disposalconstruct/attackby(obj/item/I, mob/user, params)
+	var/nicetype = get_nice_name()
+	var/ispipe = is_pipe() // Indicates if we should change the level of this pipe
+	src.add_fingerprint(user)
```
Comment (DGamerL) [inline code/modules/recycling/disposal-construction.dm:189]:
```suggestion
	add_fingerprint(user)
```

**Review by DGamerL — APPROVED** [review]
LGTM - no issues have been reported from the TM in a week or so!
Thanks for doing this!

**Review by DGamerL — APPROVED** [review]
LGTM - no issues have been reported from the TM in a week or so!
Thanks for doing this!

**Comment by DGamerL** [comment]
Thank you github for the double review

**Pull request merged**